tests/unittests/benchmark/task/test_task_wrapper.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, List
+
+import pytest
+from orion.benchmark.task.base import BaseTask
+from orion.benchmark.task.task_wrapper import FixTaskDimensionsWrapper, TaskWrapper
+from orion.core.io.space_builder import SpaceBuilder
+
+
+class DumbTask(BaseTask):
+    def __init__(self, max_trials: int, seed: int = None, a: int = 100, b: int = 10, c: int = 1):
+        super().__init__(max_trials=max_trials)
+        self.a = a
+        self.b = b
+        self.c = c
+        self.seed = seed
+
+    def call(self, x: Dict) -> List[Dict]:
+        y = self.a * x["a"] + self.b * x["b"] + self.c * x["c"]
+        return [dict(name="dumb_task", type="objective", value=y)]
+
+    def get_search_space(self) -> Dict[str, str]:
+        return {
+            "a": "uniform(0, 10, discrete=True)",
+            "b": "uniform(0, 10, discrete=True)",
+            "c": "uniform(0, 10, discrete=True)",
+        }
+
+    @property
+    def configuration(self) -> Dict[str, Any]:
+        return {
+            type(self).__qualname__: dict(
+                max_trials=self.max_trials, seed=self.seed, a=self.a, b=self.b, c=self.c,
+            )
+        }
+
+
+class TestFixDimensionsWrapper:
+    def test_init(self):
+
+        original_task = DumbTask(max_trials=10)
+        wrapped_task = FixTaskDimensionsWrapper(original_task, fixed_dims={"c": 1.0})
+
+        get_y = lambda results: results[0]["value"]
+
+        assert get_y(original_task({"a": 1, "b": 1, "c": 1})) == 111
+        assert get_y(wrapped_task({"a": 1, "b": 1, "c": 1})) == 111
+
+        assert get_y(original_task({"a": 1, "b": 1, "c": 3})) == 113
+        assert get_y(wrapped_task({"a": 1, "b": 1, "c": 3})) == 111
+
+    def test_fixed_dim_needs_to_be_part_of_space(self):
+        with pytest.raises(ValueError):
+            original_task = DumbTask(10)
+            wrapped_task = FixTaskDimensionsWrapper(original_task, fixed_dims={"foo": 1.0})
```

Comment:
Should test the error message to make sure it includes the proper dim names.

Reply:
Added a value for the `match` argument.